client/src/components/MenuHeader.tsx

```diff
@@ -1,21 +1,25 @@
-import { NavLink } from "react-router-dom";
+import { NavLink, useNavigate } from "react-router-dom";
 import { geolocaliseMe } from "../utilitiesFunctions/geolocation";
 
 export default function MenuHeader(props: {
   setters: {
     setCity: React.Dispatch<React.SetStateAction<string>>;
     setBackground: React.Dispatch<React.SetStateAction<string>>;
     setColorCircle: (value: string) => void;
+    setIdCity: (id: string) => void;
   };
 }) {
-  const { setCity, setBackground, setColorCircle } = props.setters;
+  const { setCity, setBackground, setColorCircle, setIdCity } = props.setters;
+  const navigate = useNavigate();
   return (
     <section className="menu-dropdown">
       <nav>
         <button
+          className="geoloc-button"
           type="button"
           onClick={() => {
-            geolocaliseMe(setCity, setBackground, setColorCircle);
+            geolocaliseMe(setCity, setBackground, setColorCircle, setIdCity);
+            navigate("/today"); //👉FIXE ME LATER? WARNING geolocaliseMe is not async so maybe a problem with the promise API who could arrive after the navigation
```

Comment:
Deux solutions pour résoudre le fixe me later :
- Utiliser les `loader` de react-router pour s'assurer que le fetch est bien effectuée avant l'arrivée sur le page
- Afficher une information du type `Loading in progress` sur votre page le temps que le fetch soit effectué

Reply:
Ok pour etre honnete je ne suis pas sur d'avoir le temps de le faire d'ici vendredi mais je le note en commentaire pour améliorer le projet plus tard si ça te va